tmt/steps/context/reboot.py

```diff
@@ -121,21 +136,17 @@ def handle_reboot(self, restart: Optional['RestartContext'] = None) -> bool:
 
         elif self.soft_requested:
             # Extract custom hints from the file, and reset it.
-            reboot_data = json.loads(self.request_path.read_text())
+            reboot_data = RebootData.model_validate_json(
+                self.request_path.read_text()
+            )  # TODO: Do we fail hard or handle more gracefully?
```

Comment:
Hard. There is no recovery from this: a script asked the guest to be rebooted, by calling `tmt-reboot`, and we get a broken reboot request file.